### PR TITLE
fix(helm): Change default podSecurityContext & securityContext

### PR DESCRIPTION
**Pull request by fsdaniel**
## Summary
- Update helm chart to 0.14.2
- change default podSecurityContext to set fsGroup
- set securityContext runAsUser & runAsGroup by default to match dockerfile.

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves [#2740]

### CODE REVIEW

File: helm/charts/infra/Chart.yaml [hunk 1]
```diff
@@ -3,6 +3,6 @@ name: infra
 description: A Helm chart for Infra
 type: application
 ## {x-release-please-start-version}
-version: 0.14.1
-appVersion: 0.14.1
+version: 0.14.2
+appVersion: 0.14.2
```
Comment (mxyng) [inline helm/charts/infra/Chart.yaml:7]: There's no need to increment the version. The helm chart version will be bumped automatically once a release is tagged and built.

**Review by mxyng — COMMENTED** [review]
Hey @fsdaniel, thanks for creating this PR and issue. Can you provide more context to the issue you're facing? I have not been able to reproduce the problem you described in #2740. In particular, the Kubernetes provider (minikube, EKS, GKE, etc.) and any relevant Helm values.

That being said, I see no reason not to accept this.

File: helm/charts/infra/Chart.yaml (hunk 1 above)
Comment (fsdaniel) [reply helm/charts/infra/Chart.yaml]: I do a clean helm install on a vanilla kubernetes cluster ( based on rancher RKE ). You should be able to replicate the issue by starting a clean helm install with a refresh PVC.

**Comment by fsdaniel** [comment]
> Hey @fsdaniel, thanks for creating this PR and issue. Can you provide more context to the issue you're facing? I have not been able to reproduce the problem you described in #2740. In particular, the Kubernetes provider (minikube, EKS, GKE, etc.) and any relevant Helm values.
> 
> That being said, I see no reason not to accept this.

I do a clean helm install on a vanilla on-prem Kubernetes cluster ( based on rancher RKE ). You should be able to replicate the issue by starting a clean helm install with a refresh PVC.

```yaml
server:
  service:
    type: ClusterIP
  config:
    users:
      - name: admin
        password: xxxxxxxx
    grants:
      - user: admin
        role: admin
        resource: infra
  ingress:
    enabled: true
    hosts:
      - auth.xxxxxx.io
    servicePort: 80
    className: nginx
    annotations:
      nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
      nginx.ingress.kubernetes.io/backend-protocol: "HTTP"
      nginx.ingress.kubernetes.io/proxy-http-version: "1.0"
      cert-manager.io/cluster-issuer: "letsencrypt-prod"
    tls:
      - hosts:
          - auth.xxxx.io
        secretName: io-xxx-auth
```

**Comment by mxyng** [comment]
Thanks for that info. I was able to replicate in EKS and also verify your fix. Thanks for the contribution!

**Review by mxyng — APPROVED** [review]

**Pull request merged**